tests/unit_tests/dft/source/workspace_external_tests.cpp

```diff
@@ -0,0 +1,357 @@
+    backward_t* deviceBwd = sycl::malloc_device<backward_t>(bwd_size, sycl_queue);
+    sycl_queue.copy(host_fwd.data(), device_fwd, dft_size);
+    sycl_queue.wait_and_throw();
+
+    compute_forward<decltype(desc), forward_t, backward_t>(desc, device_fwd, deviceBwd);
+    sycl_queue.wait_and_throw();
+
+    sycl_queue.copy(deviceBwd, host_bwd.data(), bwd_size);
+    sycl_queue.wait_and_throw();
+
+    // To see external workspaces, larger sizes of DFT may be needed. Using the reference DFT with larger sizes is slow,
+    // so use Parseval's theorum as a sanity check instead.
+    bool sanityCheckPasses = parseval_check(dft_size, host_fwd.data(), host_bwd.data());
+
+    if (sanityCheckPasses) {
+        sycl_queue.copy(host_fwd.data(), device_fwd, dft_size);
+        sycl_queue.wait_and_throw();
+        compute_backward<decltype(desc), backward_t, forward_t>(desc, deviceBwd, device_fwd);
+        sycl_queue.wait_and_throw();
+        sycl_queue.copy(device_fwd, host_fwd.data(), dft_size);
+        sycl_queue.wait_and_throw();
+        forward_t rescale =
+            static_cast<forward_t>(1) / static_cast<forward_t>(static_cast<scalar_t>(dft_size));
+        sanityCheckPasses = parseval_check(dft_size, host_fwd.data(), host_bwd.data(), rescale);
+    }
+
+    sycl::free(device_fwd, sycl_queue);
+    sycl::free(deviceBwd, sycl_queue);
+    sycl::free(workspace, sycl_queue);
+    return sanityCheckPasses ? !::testing::Test::HasFailure() : ::testing::Test::HasFailure();
+}
+
+template <oneapi::mkl::dft::precision prec, oneapi::mkl::dft::domain dom>
+int test_workspace_external_buffer_impl(std::size_t dft_size, sycl::device* dev) {
+    using namespace oneapi::mkl::dft;
+    using scalar_t = std::conditional_t<prec == precision::DOUBLE, double, float>;
+    using forward_t = std::conditional_t<dom == domain::COMPLEX, std::complex<scalar_t>, scalar_t>;
+    using backward_t = std::complex<scalar_t>;
+
+    sycl::queue sycl_queue(*dev);
+    if (prec == precision::DOUBLE && !sycl_queue.get_device().has(sycl::aspect::fp64)) {
+        return test_skipped;
+    }
+    descriptor<prec, dom> desc(static_cast<std::int64_t>(dft_size));
+
+    desc.set_value(config_param::WORKSPACE_PLACEMENT, config_value::WORKSPACE_EXTERNAL);
+    desc.set_value(config_param::PLACEMENT, config_value::NOT_INPLACE);
+    commit_descriptor(desc, sycl_queue);
+    std::int64_t workspace_bytes = -1;
+    desc.get_value(config_param::WORKSPACE_EXTERNAL_BYTES, &workspace_bytes);
+    if (workspace_bytes < 0) {
+        return ::testing::Test::HasFailure();
+    }
+    sycl::buffer<scalar_t> workspace(static_cast<std::size_t>(workspace_bytes) / sizeof(scalar_t));
+    desc.set_workspace(workspace);
+    // Generate data
+    std::vector<forward_t> host_fwd(static_cast<std::size_t>(dft_size));
+    std::size_t bwd_size =
+        dom == domain::COMPLEX ? dft_size : dft_size / 2 + 1; // TODO: Check this!
```

Comment:
It seems correct to me!

Reply:
TODO removed in [bee7d30](https://github.com/oneapi-src/oneMKL/pull/411/commits/bee7d3068a28fa90a868f23baab198dffad3440b).

Reply:
I don't see the change

Reply:
Well the change was in my editor... unsaved. Oops. Done in [aa0967a](https://github.com/oneapi-src/oneMKL/pull/411/commits/aa0967a974e68d6701c408969b43132fd50be684).